File: antd-js/examples/04-files.ts
/**
 * Example 04: Upload and download files and directories.
 *
 * Creates a temp file, uploads it, then downloads to a new location.
 */

import { writeFileSync, readFileSync, unlinkSync, mkdtempSync } from "node:fs";
import { join } from "node:path";
import { tmpdir } from "node:os";
import { createClient } from "../src/index.js";

const client = createClient();

// Create a temporary file to upload
const dir = mkdtempSync(join(tmpdir(), "antd-"));
const srcPath = join(dir, "test.txt");
writeFileSync(srcPath, "Hello from a file on Autonomi!");

try {
  // Estimate cost
  const cost = await client.fileCost(srcPath);
  console.log(`File upload cost estimate: ${cost} atto tokens`);

  // Upload file
  const result = await client.fileUploadPublic(srcPath);
  console.log(`File uploaded to: ${result.address}`);
  console.log(`Actual cost: ${result.cost} atto tokens`);

  // Download to new location
  const destPath = srcPath + ".downloaded";
  await client.fileDownloadPublic(result.address, destPath);
  console.log(`Downloaded to: ${destPath}`);

  const content = readFileSync(destPath, "utf-8");
  console.log(`Content: ${content}`);
  unlinkSync(destPath);
} finally {
  unlinkSync(srcPath);
}

console.log("File upload/download OK!");

File: antd-ruby/examples/04_files.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Example 04: File upload and download

require_relative "../lib/antd"

client = Antd::Client.new

# Estimate upload cost
cost = client.file_cost("/tmp/example.txt", true, false)
puts "Estimated upload cost: #{cost} atto"

# Upload a file
result = client.file_upload_public("/tmp/example.txt")
puts "File uploaded to #{result.address}"
puts "  storage: #{result.storage_cost_atto} atto, gas: #{result.gas_cost_wei} wei"
puts "  chunks: #{result.chunks_stored}, mode: #{result.payment_mode_used}"

# Download the file
client.file_download_public(result.address, "/tmp/downloaded.txt")
puts "File downloaded to /tmp/downloaded.txt"
